code/modules/research/nanites/nanite_chamber_computer.dm

```diff
@@ -0,0 +1,205 @@
+		var/obj/item/disk/nanite_program/N = I
+		if(disk)
+			eject(user)
+		if(user.transferItemToLoc(N, src))
+			to_chat(user, "<span class='notice'>You insert [N] into [src]</span>")
+			playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
+			disk = N
+	else
+		..()
+
+/obj/machinery/computer/nanite_chamber_control/proc/eject(mob/living/user)
+	if(!disk)
+		return
+	if(!istype(user) || !Adjacent(user) || !user.put_in_active_hand(disk))
+		disk.forceMove(drop_location())
+	disk = null
+
+/obj/machinery/computer/nanite_chamber_control/proc/find_chamber()
+	for(var/direction in GLOB.cardinals)
+		var/C = locate(/obj/machinery/nanite_chamber, get_step(src, direction))
+		if(C)
+			var/obj/machinery/nanite_chamber/NC = C
+			chamber = NC
+			NC.console = src
+
+/obj/machinery/computer/nanite_chamber_control/interact()
+	if(!chamber)
+		find_chamber()
+	..()
+
+/obj/machinery/computer/nanite_chamber_control/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "nanite_chamber_control", name, 550, 800, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/nanite_chamber_control/ui_data()
+	var/list/data = list()
+	if(disk)
+		data["has_disk"] = TRUE
+		var/list/disk_data = list()
+		var/datum/nanite_program/P = disk.program
+		if(P)
+			data["has_program"] = TRUE
+			disk_data["name"] = P.name
+			disk_data["desc"] = P.desc
+
+			disk_data["activated"] = P.activated
+			disk_data["activation_delay"] = P.activation_delay
+			disk_data["timer"] = P.timer
+			disk_data["activation_code"] = P.activation_code
+			disk_data["deactivation_code"] = P.deactivation_code
+			disk_data["kill_code"] = P.kill_code
+			disk_data["trigger_code"] = P.trigger_code
+			disk_data["timer_type"] = P.get_timer_type_text()
+
+			var/list/extra_settings = list()
+			for(var/X in P.extra_settings)
+				var/list/setting = list()
+				setting["name"] = X
+				setting["value"] = P.get_extra_setting(X)
+				extra_settings += list(setting)
+			disk_data["extra_settings"] = extra_settings
+			if(LAZYLEN(extra_settings))
+				disk_data["has_extra_settings"] = TRUE
+		data["disk"] = disk_data
+
+	if(!chamber)
+		data["status_msg"] = "No chamber detected."
+		return data
+
+	if(!chamber.occupant)
+		data["status_msg"] = "No occupant detected."
+		return data
+
+	var/mob/living/L = chamber.occupant
+
+	if(!(MOB_ORGANIC in L.mob_biotypes) && !(MOB_UNDEAD in L.mob_biotypes))
+		data["status_msg"] = "Occupant not compatible with nanites."
+		return data
+
+	if(chamber.busy)
+		data["status_msg"] = chamber.busy_message
+		return data
+
+	data["scan_level"] = chamber.scan_level
+	data["locked"] = chamber.locked
+	data["occupant_name"] = chamber.occupant.name
+
+	var/list/nanite_data = list()
+	SEND_SIGNAL(L, COMSIG_NANITE_GET_DATA, nanite_data)
```

Comment:
Alright I looked at the ui_data proc and it's easy enough for you to add a signal at /datum/proc/ui_data
Don't bother with making everything compatible with it for now, a standalone pr can handle the parent calls of the other ones.